packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java

```diff
@@ -14,21 +14,21 @@
 public class AssetExportDeserializer extends JsonDeserializer<AssetExport> {
 
 	@Override
-	public AssetExport deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
-		ObjectMapper mapper = (ObjectMapper) jp.getCodec();
-		JsonNode node = mapper.readTree(jp);
+	public AssetExport deserialize(final JsonParser jp, final DeserializationContext ctxt) throws IOException {
+		final ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+		final JsonNode node = mapper.readTree(jp);
 
 		final String assetTypeStr = node.get("type").asText();
 		final AssetType assetType = AssetType.getAssetType(assetTypeStr, mapper);
 
-		TerariumAsset asset = mapper.treeToValue(node.get("asset"), assetType.getAssetClass());
+		final TerariumAsset asset = mapper.treeToValue(node.get("asset"), assetType.getAssetClass());
 
 		Map<String, FileExport> files = new HashMap<>();
-		if (node.has("files")) {
+		if (node.has("files") && !node.get("files").isNull()) {
```

Comment:
What an incongrus test. How can node.files exist but return null?

Reply:
Should that be an OR and not an AND (like ContentTypeDeserializer L23)?

Reply:
Wait the logic in the two is totally confusing :). NOTs abound... .

Reply:
_ChatGPT_ says to use `if (node.hasNonNull("files")) {` instead.

Reply:
I don't understand why it worked before, but it is returning a `NullNode` entry now, makes the `has` return true, but you get an exception if you try to use the entry

Reply:
I like the `hasNonNull` ill use that instead